### PR TITLE
[REVIEW] Fix `fillna` to retain `columns` when it is `MultiIndex`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #10728 

This PR resolves the issue where `columns` can be of type `MultiIndex` and retains the column types by calling into `_from_columns_like_self` which calls the required method `_set_column_names_like`.

### CODE REVIEW

File: python/cudf/cudf/core/frame.py [hunk 1]
```diff
@@ -1124,7 +1124,7 @@ def fillna(
                 for key, value in value.items()
             }
 
-        filled_data = {}
+        columns_list = []
```
Comment (shwina) [inline python/cudf/cudf/core/frame.py:1127]: If we instead started with  a `ColumnAcccessor(multiindex=True)` here, would that have worked too?

Comment (galipremsagar) [reply python/cudf/cudf/core/frame.py]: I think `ColumnAccessor._create_unsafe` is the one we need to use here.

Comment (galipremsagar) [reply python/cudf/cudf/core/frame.py]: > I think `ColumnAccessor._create_unsafe` is the one we need to use here.

Updated code with this approach.

**Review by shwina — APPROVED** [review]
This looks like the right approach to me -- thanks, Prem!

File: python/cudf/cudf/tests/test_replace.py [hunk 2]
```diff
@@ -1371,3 +1371,17 @@ def test_replace_nulls(gsr, old, new, expected):
         expected.sort_values().reset_index(drop=True),
         actual.sort_values().reset_index(drop=True),
     )
+
+
+def test_fillna_columns_multiIndex():
```
Comment (bdice) [inline python/cudf/cudf/tests/test_replace.py:1376]: We're inconsistent about naming but we have more instances of `multiindex` than `multiIndex` in the test suite. Let's move towards using `multiindex`.
```suggestion
def test_fillna_columns_multiindex():
```

**Review by bdice — APPROVED** [review]
One minor suggestion, otherwise LGTM.

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**